src/core_modules/capture-core/components/WidgetErrorAndWarning/content/WidgetErrorAndWarningContent.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+
+import React from 'react';
+import { spacers, colors, spacersNum } from '@dhis2/ui';
+import { withStyles } from '@material-ui/core';
+import cx from 'classnames';
+import type { contentTypes, renderObjectType, renderStringType } from './WidgetErrorAndWarningContent.types';
+
+const styles = {
+    widgetWrapper: {
+        padding: `0px ${spacersNum.dp16}px`,
+    },
+    unorderedList: {
+        padding: `0px ${spacers.dp16}`,
+        marginTop: '0px',
+        lineHeight: '1.375',
+        fontSize: spacers.dp16,
+        fontWeight: 400,
+        color: colors.grey900,
+        display: 'flex',
+        flexDirection: 'column',
+        gap: spacers.dp12,
+    },
+    listItem: {
+        '&.error': {
+            listStyleType: '"!   "',
+            '&::marker': {
+                color: colors.red800,
+                fontWeight: 700,
+            },
+        },
+        '&.warning': {
+            '&::marker': {
+                color: colors.yellow900,
+                fontWeight: 700,
+            },
+        },
+    },
+};
+
+const RenderObjectItem = ({ rule, listItem }: renderObjectType) => (
+    <li
+        key={rule.id}
+        className={listItem}
+    >
+        {rule.message}
+    </li>
+);
+
+
+const RenderStringItem = ({ rule, listItem }: renderStringType) => (
+    <li
+        className={listItem}
```

Comment:
I think it's missing key in the object

Reply:
It is used directly in the return statement of the mapping, not in the function 👍

Reply:
hey @EirikHaugstulen, but it still needs a key in the item <li> (same way as the RenderObjectItem)